diff --git a/sync-all b/sync-all
index ffc150e..baf403d 100755
--- a/sync-all
+++ b/sync-all
@@ -929,24 +929,10 @@ sub main {
 
         &gitall($command, @_);
 
-        my @submodule_args = grep(/^-q/,@_);
+        my @submodule_args = @_;
 
-        if ($command eq "get") {
-            &gitInitSubmodules(@submodule_args);
-        }
-
-        if ($command eq "pull") {
-            my $gitConfig = &tryReadFile(".git/config");
-            if ($gitConfig !~ /submodule/) {
-                &gitInitSubmodules(@submodule_args);
-            }
-        }
         if ($command eq "get" or $command eq "pull") {
-            my $gitConfig = &tryReadFile(".git/config");
-            if ($gitConfig !~ /submodule/) {
-                &gitInitSubmodules(@submodule_args);
-            }
-            &git(".", "submodule", "update", @submodule_args);
+            &git(".", "submodule", "update", "--init", @submodule_args);
         }
     }
 }
