diff --git a/System/Posix/Files/Common.hsc b/System/Posix/Files/Common.hsc
index 3056014..5a2fdb2 100644
--- a/System/Posix/Files/Common.hsc
+++ b/System/Posix/Files/Common.hsc
@@ -83,15 +83,8 @@ module System.Posix.Files.Common (
 import System.Posix.Types
 import System.IO.Unsafe
 import Data.Bits
-#if defined(HAVE_STRUCT_STAT_ST_CTIM) || \
-    defined(HAVE_STRUCT_STAT_ST_MTIM) || \
-    defined(HAVE_STRUCT_STAT_ST_ATIM) || \
-    defined(HAVE_STRUCT_STAT_ST_ATIMESPEC) || \
-    defined(HAVE_STRUCT_STAT_ST_MTIMESPEC) || \
-    defined(HAVE_STRUCT_STAT_ST_CTIMESPEC)
 import Data.Int
 import Data.Ratio
-#endif
 import Data.Time.Clock.POSIX
 import System.Posix.Internals
 import Foreign.C
